File: MasterErp.Entities/Models/Item.cs
using System;
using System.ComponentModel.DataAnnotations;

#nullable disable

namespace MasterErp.Entities.Models
{
    public class Item : CreatorModel
    {
        [Key]
        public int ItemId { get; set; }
        public string NameAR { get; set; }
        public string NameEN { get; set; }
        public double? PurchasePrice { get; set; }
        public double? Yield { get; set; }
        public double Cost { get; set; }
        public int? SubUnitId { get; set; }
        public int MainUnitId { get; set; }
        public int ItemCategoryId { get; set; }
        public double? ConvertRatio { get; set; }
        public bool IsActive { get; set; }
        public int? ItemType { get; set; }
    }
}

File: MasterErp.Angular/src/app/components/Inventory/models/Item.ts
export interface ItemModel {
    itemId?: number;
    nameEN?: string;
    nameAR?: string;
    subUnitId?: number;
    mainUnitId?: number;
    itemCategoryId?: number;
    cost?: number;
    convertRatio?: number;
    isActive?: boolean;
    insertUser?: string;
    insertDate?: string;
    itemsSupplier?: ItemSupplier[];
}

export interface ItemSupplier {
    id?: number;
    supplierId?: number;
    supplierName?: string;
    itemId?: number;
    itemName?: string;

}